app-modules/portal/src/Filament/Pages/ManagePortalSettings.php

```diff
@@ -139,6 +145,52 @@ public function form(Form $form): Form
                             ->hintIconTooltip('Surveys are not a part of your current subscription.'),
                     ])
                     ->columns(3),
+                Section::make('Knowledge Portal')
+                    ->schema([
+                        Toggle::make('knowledge_base_portal_enabled')
+                            ->label('Knowledge Management')
+                            ->disabled(! Gate::check(Feature::KnowledgeManagement->getGateName()))
+                            ->hintIcon(fn (Toggle $component) => $component->isDisabled() ? 'heroicon-m-lock-closed' : null)
+                            ->hintIconTooltip('Knowledge Management is not a part of your current subscription.')
+                            ->live()
+                            ->columnSpanFull(),
+                        ColorPicker::make('knowledge_base_portal_primary_color')
+                            ->label('Primary Color')
+                            ->hexColor()
+                            ->visible(fn (Get $get) => $get('knowledge_base_portal_enabled'))
+                            ->columnSpan(1),
+                        Select::make('knowledge_base_portal_rounding')
+                            ->label('Rounding')
+                            ->options(Rounding::class)
+                            ->visible(fn (Get $get) => $get('knowledge_base_portal_enabled'))
+                            ->columnSpan(1),
+                        TextInput::make('knowledge_base_portal_authorized_domain')
+                            ->label('Authorized Domain')
+                            ->url()
+                            ->visible(fn (Get $get) => $get('knowledge_base_portal_enabled'))
+                            ->columnSpanFull(),
+                        Toggle::make('knowledge_base_portal_service_management')
+                            ->label('Service Management')
+                            ->visible(fn (Get $get) => $get('knowledge_base_portal_enabled'))
+                            ->disabled(! Gate::check(Feature::ServiceManagement->getGateName()))
+                            ->hintIcon(fn (Toggle $component) => $component->isDisabled() ? 'heroicon-m-lock-closed' : null),
```

Comment:
If the account previously had the feature and had `knowledge_base_portal_enabled` set to `true`, would they still be able to edit all these other settings?

And if so, should they? Curious what your thoughts on that are and whether we need to ask Product.

Right now I am leaning towards: they should be visible based on the check you have now. But also have the feature lock and tooltip to disable.

Reply:
@Orrison I will update this in the subsequent PR for the Portal.